Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/openSAML/extraction/OpenSAMLAssertionExtractorImpl.java

```diff
@@ -176,66 +176,88 @@ private void populateAttributeStatement(final Assertion saml2Assertion, final As
-                    }
-
-                } else if (attribute.getName().equals(NhincConstants.ATTRIBUTE_NAME_NPI)) {
-                    String nationalProviderId = getAttributeValue(attribute);
-                    target.setNationalProviderId(nationalProviderId);
-                    LOG.debug("Assertion.nationalProviderId = " + nationalProviderId);
-
-                } else {
-                    LOG.warn("Unrecognized Name Attribute: " + attribute.getName());
+    /**
+     * @param attribute
+     * @param target
+     * @param attrName
+     */
+    private void compareAttribute(Attribute attribute, AssertionType target, AttributeHelper helper) {
+
+        if (attribute.getName().equals(NhincConstants.ATTRIBUTE_NAME_SUBJECT_ROLE)) {
+            LOG.debug("Extracting Assertion.userInfo.roleCoded:");
+            populateSubjectRole(attribute, target);
+
+        } else if (attribute.getName().equals(NhincConstants.ATTRIBUTE_NAME_PURPOSE_OF_USE)) {
+            LOG.debug("Extracting Assertion.purposeOfDisclosure:");
+            populatePurposeOfUseAttribute(attribute, target);
+
+        } else if (attribute.getName().equals(NhincConstants.USERNAME_ATTR)) {
+            helper.extractNameParts(attribute, target);
+
+        } else if (attribute.getName().equals(NhincConstants.USER_ORG_ATTR)) {
+            String orgAttribute = getAttributeValue(attribute);
+            target.getUserInfo().getOrg().setName(orgAttribute);
+            LOG.debug("Assertion.userInfo.org.Name = {}", orgAttribute);
+
+        } else if (attribute.getName().equals(NhincConstants.USER_ORG_ID_ATTR)) {
+            String orgIDAttribute = getAttributeValue(attribute);
+            target.getUserInfo().getOrg().setHomeCommunityId(orgIDAttribute);
+            LOG.debug("Assertion.userInfo.org.homeCommunityId = {}", orgIDAttribute);
+
+        } else if (attribute.getName().equals(NhincConstants.ATTRIBUTE_NAME_HCID)) {
+            String homeCommunityId = getAttributeValue(attribute);
+            target.getHomeCommunity().setHomeCommunityId(homeCommunityId);
+            LOG.debug("Assertion.homeCommunity.homeCommunityId = {}", homeCommunityId);
+
+        } else if (attribute.getName().equals(NhincConstants.ACCESS_CONSENT_ATTR)) {
+            List<String> accessConsentId = transformXMLtoString(attribute.getAttributeValues());
+            target.getSamlAuthzDecisionStatement().getEvidence().getAssertion().getAccessConsentPolicy()
+            .addAll(accessConsentId);
+            LOG.debug(
+                "Assertion.SamlAuthzDecisionStatement.Evidence.Assertion.AccessConsentPolicy = {}", accessConsentId);
+
+        } else if (attribute.getName().equals(NhincConstants.INST_ACCESS_CONSENT_ATTR)) {
+            List<String> instAccessConsentId = transformXMLtoString(attribute.getAttributeValues());
+            target.getSamlAuthzDecisionStatement().getEvidence().getAssertion().getInstanceAccessConsentPolicy()
+            .addAll(instAccessConsentId);
```

Comment:
Was this run through one of the CONNECT IDE formatters?  I'm not sure if would have caught all the extra newlines in this class, but I suspect this line would have been indented automatically.